File: frontend/productivity/src/Task.js
class Task {
	constructor() {
		this.id = undefined;
		this.projectId = undefined;
		this.isBeingWorkedOn = false;
		this.isDeleted = false;
		this.title = "";
		this.description = {
			shortDescription: "",
			longDescription: "",
			additionalNotes: undefined
		};
        this.endDate = undefined;
        this.priority = 0;
        this.parentTasks = undefined;
		this.ui = {
			xposition: 5,
			yposition: 45
		};
	}

	static from(json){
		return Object.assign(new Task(), json);
	  }
}

export default Task;

File: src/main/kotlin/de/nykon/productivity/domain/value/Task.kt
package de.nykon.productivity.domain.value

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.index.Indexed
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDateTime
import java.util.*

/**
 * A single task.
 */
@Document(collection = "tasks")
data class Task(
    @Id val id: String = UUID.randomUUID().toString(),
    @Indexed val projectId: String?,
    val isBeingWorkedOn: Boolean = false,
    val isDeleted: Boolean = false,
    val title: String,
    val description: TaskDescription,
    val deadline: String? = null,
    @Indexed val notifyRelativeDate: String? = null,
    @Indexed val notifyDateBeforeDeadline: String? = null,
    val priority: Int = 0,
    val parentTasks: List<String>? = null,
    val ui: UI = UI(25,25),
    val createTime: LocalDateTime = LocalDateTime.now()

)